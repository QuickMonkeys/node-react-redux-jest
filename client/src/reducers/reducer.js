import Immutable from 'immutable';

const initialState = Immutable.Map ({
                        languages: [],
                        filter: undefined
                    });

export default (state = initialState, action) =>
{
    switch (action.type) {

        case "POPULATE":
            return state.set('languages', action.value);

        case "ALL":
            return state.set('filter', undefined);
            
        case "FILTER":
            return state.set('filter', action.value);

        default: {
            return  state;
        }
    }
}