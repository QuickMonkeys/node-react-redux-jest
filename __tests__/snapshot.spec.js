'use strict';

import React from 'react';
import { createStore } from 'redux';
import { Provider } from  'react-redux';
import renderer from 'react-test-renderer';

import Search from '../client/src/components/search.jsx';
import Languages from '../client/src/containers/languages.js';

import reducer from '../client/src/reducers/reducer.js';

describe('Generate snapshots from the components', () => {

    it('Search', () => {
        const obj = renderer.create(<Search />).toJSON();
        expect(obj).toMatchSnapshot();
    });

    it('Languages', () => {
        window.$ = require('jquery');
        const store = createStore(reducer);
        const obj = renderer.create(<Provider store={store}>
                                        <Languages />
                                    </Provider>).toJSON();
        expect(obj).toMatchSnapshot();
    });
});
