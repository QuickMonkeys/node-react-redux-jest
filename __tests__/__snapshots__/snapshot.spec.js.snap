exports[`Generate snapshots from the components Languages 1`] = `
<table
  className="table">
  <thead>
    <tr>
      <th
        colSpan="3">
        Total: 
        0
         of 
        0
      </th>
    </tr>
    <tr>
      <th>
        Position
      </th>
      <th>
        Language
      </th>
      <th>
        Rating
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        colSpan="3"
        style={
          Object {
            "textAlign": "center",
          }
        }>
        Loading...
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Generate snapshots from the components Search 1`] = `
<div>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search for ..."
    type="text" />
</div>
`;
